pkg/cmd/clients.go

```diff
@@ -130,68 +134,165 @@ func (cc *ClientCmd) CreateClientCmd() *cobra.Command {
+			if err := json.Unmarshal(extMeta, &extServiceClass); err != nil {
+				return errors.WithStack(err)
+			}
+
+			si := v1beta1.ServiceInstance{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "servicecatalog.k8s.io/v1beta1",
+					Kind:       "ServiceInstance",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace:    namespace,
+					GenerateName: validServiceName + "-",
+				},
+				Spec: v1beta1.ServiceInstanceSpec{
+					PlanReference: v1beta1.PlanReference{
+						ClusterServiceClassExternalName: clusterServiceClass.Spec.ExternalName,
+					},
+					ClusterServiceClassRef: &v1beta1.ClusterObjectReference{
+						Name: clusterServiceClass.Name,
+					},
+					ClusterServicePlanRef: &v1beta1.ClusterObjectReference{
+						Name: "default",
+					},
+					ParametersFrom: []v1beta1.ParametersFromSource{
+						{
+							SecretKeyRef: &v1beta1.SecretKeyReference{
+								Name: clientId + "-apb-" + "params",
+								Key:  "parameters"},
+						},
+					},
+				},
+			}
+
+			if _, err := cc.scClient.ServicecatalogV1beta1().ServiceInstances(namespace).Create(&si); err != nil {
+				return errors.WithStack(err)
+			}
+			fmt.Println("creating service")
+			pSecret := v1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: clientId + "-apb-" + "params",
+				},
+			}
+			pSecret.Data = map[string][]byte{}
+
+			parameters := map[string]string{}
+			parameters["appName"] = name
+			parameters["appIdentifier"] = appIdentifier
+			secretData, err := json.Marshal(parameters)
+			if err != nil {
+				return errors.WithStack(err)
+			}
+			pSecret.Data["parameters"] = secretData
+			if _, err := cc.k8Client.CoreV1().Secrets(namespace).Create(&pSecret); err != nil {
+				return errors.WithStack(err)
+			}
+
+			noWait, err := cmd.PersistentFlags().GetBool("no-wait")
+			if err != nil {
+				return errors.WithStack(err)
+			}
+			if noWait {
+				return nil
+			}
+			timeout := int64(5 * 60) // ten minutes
+			w, err := cc.scClient.ServicecatalogV1beta1().ServiceInstances(namespace).Watch(metav1.ListOptions{TimeoutSeconds: &timeout})
+			if err != nil {
+				return errors.WithStack(err)
+			}
+			for {
+				select {
+				case msg, ok := <-w.ResultChan():
+					if !ok {
+						fmt.Println("Timedout waiting. It seems to be taking a long time for the service to provision. Your service may still be provisioning.")
+						return nil
+					}
+					o := msg.Object.(*v1beta1.ServiceInstance)
+					switch msg.Type {
+					case watch.Error:
+						w.Stop()
+						return errors.New("unexpected error watching ServiceInstance " + err.Error())
+					case watch.Modified:
+						for _, c := range o.Status.Conditions {
+							fmt.Println("status: " + c.Message)
+							if c.Type == "Ready" && c.Status == "True" {
+								w.Stop()
+								return nil
+							}
+
+							if c.Type == "Failed" {
+								w.Stop()
+								return errors.New("Failed to provision " + extServiceClass.ServiceName + ". " + c.Message)
+							}
+						}
```

Comment:
@witmicko I' unsure about this behaviour because it doesn't seem consistent with other cli behaviour from oc or kubectl. IMO the request should just end with error or ok if the server was happy to accept and the user can initiate a watch later if they wish but there may be reasons for the above, dont have enough context to say if this should be removed or remain

Reply:
We do this for other services and allow a --no-wait flag to be passed. So I think this is ok for now. It may be something we want to change where no-wait is the default and you can pass an option to wait. But not something for this issue I think

Reply:
> We do this for other services

@witmicko, @maleck13  Cool, fine with this then if its consistent with others